regcomp.c

```diff
@@ -19291,20 +19408,20 @@ S_regclass(pTHX_ RExC_state_t *pRExC_state, I32 *flagp, U32 depth,
 
                     /* Resolve the run-time inversion flag with this possibly
                      * inverted class */
-                    invert = invert ^ already_inverted;
+                    *invert = *invert ^ already_inverted;
 
-                    ret = reg_node(pRExC_state,
-                                   POSIXL + invert * (NPOSIXL - POSIXL));
-                    FLAGS(REGNODE_p(ret)) = classnum;
-                    goto not_anyof;
+                    op = POSIXL + *invert * (NPOSIXL - POSIXL);
```

Comment:
Yes, if this is done, It shouldn't be in this commit, which is trying to avoid doing more than necessary in splitting the function.

Your proposal is clearer.   A reason not to do what you suggest is because this is a repeat of a common paradigm in this file, so it isn't out of the blue, and it works without change if a new type of POSIXL regnode (and inverse) is added, which has long been a possibility and remains so.